core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -150,14 +192,40 @@ private void cacheManifests() {
   }
 
   @Override
+  public List<ManifestFile> allManifests(FileIO fileIO) {
+    if (allManifests == null) {
+      cacheManifests(fileIO);
+    }
+    return allManifests;
+  }
+
+  /**
+   * @deprecated since 1.0.0 - Use {@link Snapshot#allManifests(FileIO)} instead.
+   */
+  @Override
+  @Deprecated
   public List<ManifestFile> allManifests() {
+    // TODO - Consider throwing here if allManifests are not already cached or if called at all.
     if (allManifests == null) {
       cacheManifests();
     }
```

Comment:
@rdblue and I discussed leaving this not deprecated, but throwing if the function _needs_ to call `cacheManifests()`.

I'd personally like to see the older signatures removed prior to 1.0, but given we're not doing a 0.14.0 release that might not be possible.

However, we should be doing as much as possible to discourage people from using the function signatures that don't require `FileIO`. It might be ok to keep them for backwards compatibility, provided that `cacheManifests` and `cacheChanges` have already been called.

Reply:
I think I'm okay with deprecating this.

Reply:
Yes same. I left a note that it's deprecated and will be removed in 1.0.0.